app/controllers/api/v1/upload_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Api
+  module V1
+    class UploadController < ApplicationController
+      def create
+        upload = Upload.new(
+          search_engine: params[:search_engine],
+          csv_file: params[:csv_file]
+        )
+        if upload.save
+          render json: { message: I18n.t('upload.upload_success'), status: :ok }
```

Comment:
1. It would be better to follow [JSON API spec](https://jsonapi.org/) so with `data` attribute. 
2. We generally use serializer to render API responses (`app/serializers`). Past project can provide an example.
3. When creating data, the status code should be `201`: http://www.railsstatuscodes.com/created.html#:~:text=201%20%2D%20Created&text=The%20request%20has%20been%20fulfilled,a%20new%20resource%20being%20created.

Reply:
Since we're returning a simple success message, is it necessary to follow step 1 & 2? 💭

Reply:
On creation, the API should return a success message and the created resources. It is a best practice. So yes following the steps 1 and 2 are needed.

Reply:
Since you're processing bulk import, you can return the total keywords in the response 💭